src/rust/engine/fs/src/snapshot.rs

```diff
@@ -341,6 +341,66 @@ impl Snapshot {
       .to_boxed()
   }
 
+  pub fn strip_prefix(
+    store: Store,
+    root_digest: Digest,
+    prefix: PathBuf,
+  ) -> impl Future<Item = Digest, Error = String> {
+    let store2 = store.clone();
+    Self::get_directory_or_err(store.clone(), root_digest)
+      .and_then(move |dir| {
+        futures::future::loop_fn(
+          (dir, PathBuf::new(), prefix),
+          move |(dir, already_stripped, prefix)| {
+            let mut components = prefix.components();
+            let component_to_strip = components.next();
+            if let Some(component_to_strip) = component_to_strip {
+              let remaining_prefix = components.collect();
+              let index = dir
+                .get_directories()
+                .binary_search_by(|node| {
+                  node
+                    .get_name()
+                    .cmp(component_to_strip.as_os_str().to_string_lossy().as_ref())
+                })
+                .map_err(|_index| {
+                  // TODO: Should this error, or just return an empty Directory?
```

Comment:
I think that erroring is the right behavior _except_ in the case of an already empty directory. We should just succeed silently in the case of an empty directory.